### PR TITLE
create event based request

### PR DESCRIPTION
**Pull request by Schparky**
image

image

image

image

### CODE REVIEW

File: src/components/CreateOrUpdateRequest.svelte [hunk 1]
```diff
@@ -88,8 +81,14 @@ async function cancelRequest() {
   cancelled()
 }
 
+function onEventChange(e) {
+  request.event_id = e.detail
+  request.destination = $events.find(({ id }) => id === e.detail).location
```
Comment (hobbitronics) [inline src/components/CreateOrUpdateRequest.svelte:86]: not sure, but are the () needed around the object parameter?

Comment (Schparky) [reply src/components/CreateOrUpdateRequest.svelte:86]: Good question. I just tried without...no good

File: src/components/Request.svelte [hunk 2]
```diff
@@ -67,13 +67,16 @@ function goToConversation(conversationId) {
...
-              {:else}
-                { destination }
-              {/if}
+              { destination }
             </dd>
 
+            {#if request.meeting}
+              <dt class={dlTermColumns}>Event</dt>
+              <dd class={dlDescriptionColumns}>
+                {request.meeting.name}
```
Comment (hobbitronics) [inline src/components/Request.svelte:76]:
```suggestion
                {request.meeting?.name}
```

File: src/components/RequestListEntry.svelte [hunk 3]
```diff
@@ -9,7 +9,7 @@ export let request;
 $: user = request.created_by || {}
 $: size = request.size
 $: from = request.origin && request.origin.description
-$: to = request.meeting ? request.meeting.name : request.destination.description
+$: to = request.destination.description
```
Comment (hobbitronics) [inline src/components/RequestListEntry.svelte:12]:
```suggestion
$: to = request.destination?.description
```
Not sure if there is really a case where you will get a request without destination or description

**Review by hobbitronics — APPROVED** [review]

**Pull request merged**